crates/host/src/module.rs

```diff
@@ -197,10 +197,21 @@ impl ModuleCache {
         match self.get_from_filesystem_cache(key) {
             // Filesystem cache hit, deserialize and save to deserialized cache
             Ok(Some(serialized_module)) => {
-                let module = Arc::new(
-                    unsafe { Module::deserialize(&self.runtime_engine, serialized_module.clone()) }
-                        .map_err(|e| wasm_error!(WasmErrorInner::ModuleBuild(e.to_string())))?,
-                );
+                let module_result =
+                    unsafe { Module::deserialize(&self.runtime_engine, serialized_module.clone()) };
+
+                // If deserialization fails, we assume the file is corrupt,
+                // so it is removed from the filesystem cache.
+                let module = match module_result {
+                    Ok(d) => Ok(Arc::new(d)),
+                    Err(e) => {
+                        let _ = self.remove_from_filesystem_cache(key);
```

Comment:
Remove from cache, but ignore any errors deleting the file (i.e. if it doesn't exist or user doesn't have permissions). Not sure if the error should be passed up.